### PR TITLE
Tommy/object and player collisions

### PR DESCRIPTION
**Pull request by teelau**

### CODE REVIEW

File: server/game/game.go [hunk 1]
```diff
@@ -63,14 +63,10 @@ func (a *Arena) UpdatePositions() {
...
-	// for _, player := range a.Players {
-	// 	// TODO: Player to Player
-	// 	// TODO: Player to Junk
-	// }
-	// for _, hole := range a.Holes {
-	// 	// TODO: Hole to Player
-	// 	// TODO: Hole to Junk
-	// }
+	a.collisionPlayerToPlayer()
+	a.collisionPlayerToJunk()
```
Comment (brian-nguyen) [inline server/game/game.go]: You loop over players twice

File: server/game/game.go [hunk 2]
```diff
@@ -63,14 +63,10 @@ func (a *Arena) UpdatePositions() {
...
+	a.collisionHoleToJunk()
+	a.collisionHoleToPlayer()
```
Comment (brian-nguyen) [inline server/game/game.go]: Same here. Holes are looped over twice

File: server/game/game.go (hunk 1 above)
Comment (teelau) [reply server/game/game.go]: wouldn't they take the same time as a single loop?
we can also split them into separate go routines

File: server/game/game.go [hunk 3]
```diff
@@ -126,16 +126,57 @@ func (a *Arena) collisionPlayerToPlayer() {
...
+				//Player falls into hole
+				//TODO: implement events for player falling into hole
 			}
+		}
+	}
+}
 
+func (a *Arena) collisionHoleToJunk() {
+	for _, hole := range a.Holes {
+		for _, player := range a.Players {
```
Comment (brian-nguyen) [inline server/game/game.go]: This should be looping over Junk not Players

File: server/game/game.go [hunk 4]
```diff
@@ -126,16 +126,57 @@ func (a *Arena) collisionPlayerToPlayer() {
...
+
+func (a *Arena) collisionPlayerToJunk() {
+	for _, player := range a.Players {
+		for _, junk := range a.Junk {
+			if areCirclesColliding(player.Position, playerRadius, junk.Position, junkRadius) {
+				//Junk calls player.hitJunk function to calculate player state
+				junk.HitBy(&player)
+				//Assign junk to last recently hit player color/id
+				junk.ID = player.ID
+				junk.Color = player.Color
```
Comment (brian-nguyen) [inline server/game/game.go]: The HitBy function should handle setting the color and player id

**Pull request merged**